integration/config/config_test.go

```diff
@@ -267,19 +268,32 @@ func TestTemplatedConfig(t *testing.T) {
 		swarm.ServiceWithName("svc"),
 	)
 
-	var tasks []swarmtypes.Task
-	waitAndAssert(t, 60*time.Second, func(t *testing.T) bool {
-		tasks = swarm.GetRunningTasks(t, d, serviceID)
-		return len(tasks) > 0
-	})
+	var task swarmtypes.Task
+	getServiceTask := func(log poll.LogT) poll.Result {
+		filter := filters.NewArgs()
+		filter.Add("desired-state", "running")
+		filter.Add("service", serviceID)
+		tasks, err := client.TaskList(context.Background(), types.TaskListOptions{
+			Filters: filter,
+		})
 
-	task := tasks[0]
-	waitAndAssert(t, 60*time.Second, func(t *testing.T) bool {
-		if task.NodeID == "" || (task.Status.ContainerStatus == nil || task.Status.ContainerStatus.ContainerID == "") {
-			task, _, _ = client.TaskInspectWithRaw(context.Background(), task.ID)
+		switch {
+		case err != nil:
+			return poll.Error(err)
+		case len(tasks) > 0:
+			task = tasks[0]
+			if task.NodeID == "" || (task.Status.ContainerStatus == nil || task.Status.ContainerStatus.ContainerID == "") {
+				task, _, _ = client.TaskInspectWithRaw(context.Background(), task.ID)
+			}
+			if task.NodeID != "" && task.Status.ContainerStatus != nil && task.Status.ContainerStatus.ContainerID != "" {
+				return poll.Success()
+			}
+			return poll.Continue("task still waiting")
+		default:
+			return poll.Continue("task still waiting")
 		}
-		return task.NodeID != "" && task.Status.ContainerStatus != nil && task.Status.ContainerStatus.ContainerID != ""
-	})
+	}
+	poll.WaitOn(t, getServiceTask, swarm.ServicePoll, poll.WithTimeout(1*time.Minute))
```

Comment:
I wonder if we should split this into two steps;

- check if the service reconciled (check if the desired number of tasks is running); that may be more generic / reusable
- once that's the case, do the `inspect` on its task